### PR TITLE
CLI to provision via pgp, signup push pgp

### PR DESCRIPTION
**Pull request by patrickxb**
1. Add `--pgp` flag to devel `signup` command that pushes a PGP key to API server
2. Add devel command `pgpprovision` that will provision a new device with synced PGP key

r? @maxtaco

### CODE REVIEW

**Comment by patrickxb** [comment]
@keybase/react-hackers I need some flow fixes here...thanks!

File: go/client/cmd_pgpprovision.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	username   string
+	deviceName string
+	passphrase string
+}
+
+func NewCmdPGPProvision(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "pgpprovision",
+		Usage:        "Provision a device via PGP",
+		ArgumentHelp: "[username] [passphrase] [devicename]",
```
Comment (maxtaco) [inline go/client/cmd_pgpprovision.go]: just to double-check -- this is dev-only, right?

Comment (patrickxb) [inline go/client/cmd_pgpprovision.go]: yes, it's only in commands_devel.go and the RPC won't run in production.

File: go/engine/pgpprovision.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	// need a session to try to get synced private key
+	return e.G().LoginState().LoginWithPassphrase(e.user.GetName(), e.passphrase, false, afterLogin)
+}
+
+// syncedPGPKey looks for a synced pgp key for e.user.  If found,
+// it unlocks it.
+func (e *PGPProvision) syncedPGPKey(ctx *Context) (libkb.GenericKey, error) {
+	key, err := e.user.SyncedSecretKey(ctx.LoginContext)
```
Comment (maxtaco) [inline go/engine/pgpprovision.go]: weird white space here....

**Comment by maxtaco** [comment]
👍

**Comment by chrisnojima** [comment]
fixing

**Pull request merged**